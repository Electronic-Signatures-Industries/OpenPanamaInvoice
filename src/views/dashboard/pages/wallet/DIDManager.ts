import { DIDDocument, Wallet } from 'xdvplatform-wallet';
import { KeystoreIndex } from '../shared/KeystoreIndex';
import { Session } from '../shared/Session';
import { CID, IpfsHttpClient } from 'ipfs-http-client'
// const ipfs = IpfsHttpClient('https://ipfs-api.xdv.digital/api/v0')
import { Ed25519Provider } from 'key-did-provider-ed25519'
import KeyResolver from '@ceramicnetwork/key-did-resolver'
import { DID } from 'dids'
import { mnemonicToSeed } from 'ethers/utils/hdnode';
import { arrayify } from 'ethers/utils';

export class DIDManager {

    async create3ID(wallet: Wallet) {
        const provider = new Ed25519Provider(arrayify(mnemonicToSeed(wallet.mnemonic)))
        const did = new DID({ provider, resolver: KeyResolver.getResolver() })
        await did.authenticate()
        return did;
    }
    
    async createDID(
        ks: KeystoreIndex,
        wallet: Wallet,
        ipfs: any,
        messageNotify: (m) => {}
    ) {

        const keypairExports = await wallet.getPrivateKeyExports("P256");
        const keypairEd25519Exports = await wallet.getPrivateKeyExports('ED25519');
        messageNotify("Connecting to Swarm");
        // const swarmFeed = await wallet.getSwarmNodeClient(
        //   ks.address,
        //   "ES256K"
        //   // 'https://ipfs.auth2factor.com/'
        // );
        const session = `did:ethr:${ks.address}`;
        const did = new DIDDocument();
        const pub = { owner: ks.address, ...keypairExports.ldJsonPublic };
        pub.publicKeyJwk.d = undefined;
        did.id = session;
        did.publicKey = [pub];
        did.authentication = [pub as any];
        const didIndex = { ...did, tag: "main_did" };


        messageNotify("Requesting access to publish...");

        // const file = await ipfs.add(Buffer.from(JSON.stringify(didIndex)))
        // const cid = file.cid();
        // // static
        // const res = await ipfs.name.publish(`/ipfs/${cid}`);

        // const key = await ipfs.key.import('ed25519', keypairEd25519Exports.pem, '');
        // debugger
        const file = await ipfs.add(Buffer.from(JSON.stringify(didIndex)))
        const cid = file.path;
        // static
        const res = await ipfs.name.publish(`/ipfs/${cid}`);

        console.log(res);
        // const directory = {
        //     'file': { data: Buffer.from('nice essay') },
        //     'other-file': { data: Buffer.from('nice essay') },
        //     'folder/with-file': { data: Buffer.from('nice essay') },
        //     'folder/with-other-folder/and-file': { data: Buffer.from('nice essay') }
        // }
        // const rootHash = await storage.put(directory)
        // const retrievedDirectory = await storage.get(rootHash)
        Session.set({ ks });

        return res
    }

}