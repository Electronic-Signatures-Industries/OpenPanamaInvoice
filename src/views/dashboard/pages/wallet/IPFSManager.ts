import IPFS from 'ipfs-core'
import dagJose from 'dag-jose'
import multiformats from 'multiformats/cjs/src/basics'
import legacy from 'multiformats/cjs/src/legacy'
import { DID } from 'dids'
import { keccak256 } from 'ethers/utils'
import { ethers } from 'ethers'
import moment from 'moment'

const mf = multiformats('');
mf.multicodec.add(dagJose)
const dagJoseFormat = legacy(mf, dagJose.name)

export class IPFSManager {
    client: IPFS.IPFSRepo;
    provider: ethers.providers.JsonRpcProvider
    async start() {
        this.provider = new ethers.providers.JsonRpcProvider('https://mainnet.infura.io/');
        this.client = await IPFS.create({ ipld: { formats: [dagJoseFormat] } })
    }

    async stop() {
        await this.client.stop();
    }

    /**
     * Converts Blob to Keccak 256 hash
     * @param payload 
     */
    async blobToKeccak256(payload: Blob): Promise<string> {
        let ab = await payload.arrayBuffer();
        let buf = new Uint8Array(ab);
        return keccak256(buf) as string;
   }

   async setCurrentNode(
       did: DID,
       cid: string,       
   ) {
    debugger   
    // TODO: Use IPNS
       const res = await this.client.name.publish(cid);

       debugger
       return res;
   }

    /**
     * Add Signed Object
     * @param did DID
     * @param payload Payload, either Buffer or Blob 
     * @param previousNode If it has previous node
     */
    async addSignedObject(
        did: DID,
        payload: Buffer | Blob,
        previousNode: any) {
        let temp: string;
        let content: Buffer;
        if (payload instanceof Blob) {
            temp = await this.blobToKeccak256(payload);
            content = Buffer.from((await payload.arrayBuffer()));
        } else {
            temp = keccak256(payload);
            content = payload;
        }
        temp = temp.replace('0x','');
        const epoch = 1; // await this.provider.getBlockNumber();
        // sign the payload as dag-jose
        const { jws, linkedBlock } = await did.createDagJWS({
            epoch,
            timestamp: moment().unix(),
            hash: temp,
            content: content.toString('hex'),
            parent: previousNode || undefined
        });
        // put the JWS into the ipfs dag
        const jwsCid = await this.client.dag.put(jws, { format: 'dag-jose', hashAlg: 'sha2-256' })
        // put the payload into the ipfs dag
        await this.client.block.put(linkedBlock, { cid: jws.link })
        return jwsCid.toString()
    }

    /**
     * Get IPLD object
     * @param cid content id
     */
    async getObject(cid: string): Promise<any> {
        let temp = await this.client.dag.get(cid);
        const res = {
            metadata: {
                ...temp,
            },            
            payload: undefined            
        }
        temp = await this.client.dag.get(cid, { path: '/link' });
        res.payload = {
            ...temp,
        };

        return temp;
    }

    verify(did: DID, obj: any): Promise<any> {
        return did.verifyJWS(obj.metadata);
    }

    async encryptObject(did: DID, cleartext, dids: string[]) {
        const jwe = await did.createDagJWE(cleartext, dids)
        return this.client.dag.put(jwe, { format: 'dag-jose', hashAlg: 'sha2-256' })
    }

    async decryptObject(did: DID, cid, query) {
        const jwe = (await this.client.dag.get(cid, query)).value
        const cleartext = await did.decryptDagJWE(jwe)
        return cleartext;
    }
    
}