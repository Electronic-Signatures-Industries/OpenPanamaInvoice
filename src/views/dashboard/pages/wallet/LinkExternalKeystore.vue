<template>
  <v-content>
    <v-dialog v-model="showSCLogin" max-width="500px">
      <v-card>
        <v-card-title>
          <span class="headline">Enter smartcard PIN</span>
        </v-card-title>

        <v-card-text>
          <v-form autocomplete="off">
            <v-row>
              <v-col cols="12" md="12">
                <v-text-field
                  required
                  v-model="pin"
                  :append-icon="showPassword ? 'mdi-eye' : 'mdi-eye-off'"
                  :type="showPassword ? 'text' : 'password'"
                  class="input-group--focused"
                  @click:append="showPassword = !showPassword"
                  :error="!!validations.password"
                  :hint="validations.password"
                ></v-text-field>
              </v-col>
            </v-row>
          </v-form>
        </v-card-text>

        <v-card-actions>
          <v-spacer></v-spacer>
          <v-btn color="blue darken-1" text @click="showSCLogin = false"
            >Cancel</v-btn
          >
          <v-btn color="blue darken-1" text @click="loginSmartCard">OK</v-btn>
        </v-card-actions>
      </v-card>
    </v-dialog>
    <v-dialog v-model="show" max-width="600px">
      <v-card>
        <v-card-title>
          <span class="headline">Link external hardware / software module</span>
        </v-card-title>

        <v-card-text>
          <v-form autocomplete="off">
            <!-- <v-row>
              <v-col cols="12" md="12">
                <v-radio-group
                  v-model="value.type"
                  class="font-weight-medium"
                  label="Type"
                  column
                >
                <v-tooltip bottom>
                    <template v-slot:activator="{ on }">
                      <v-radio
                        v-on="on"
                        class="font-weight-medium"
                        label="WalletConnect"
                        color="red"
                        value="walletconnect"
                      ></v-radio>
                    </template>
                    <span>WalletConnect</span>
                  </v-tooltip> 
                  <v-tooltip bottom>
                    <template v-slot:activator="{ on }">
                      <v-radio
                        v-on="on"
                        class="font-weight-medium"
                        label="Ledger"
                        color="red"
                        value="ledger"
                      ></v-radio>
                    </template>
                    <span>Ledger</span>
                  </v-tooltip>
                  <v-tooltip bottom>
                    <template v-slot:activator="{ on }">
                      <v-radio
                        v-on="on"
                        class="font-weight-medium"
                        label="PKCS#11 Smart Card"
                        color="red"
                        value="pkcs11"
                      ></v-radio>
                    </template>
                    <span>PKCS#11 Smart card</span>
                  </v-tooltip>
                </v-radio-group>
              </v-col>
            </v-row>
             -->
            <v-row>
              <v-col cols="12" md="12">
                <v-select
                  persistent-hint
                  hint="DID signer"
                  required
                  :items="defaultDIDSignerOptions"
                  v-model="value.defaultDIDSigner"
                  item-text="key"
                  class="font-weight-medium"
                  item-value="value"
                  @input="setDefaultSignerProtocol"
                  label="Select default DID signer"
                  single-line
                ></v-select>
              </v-col>

              <v-col cols="12" md="12">
                <v-select
                  persistent-hint
                  hint="X509 signer"
                  required
                  :items="defaultX509SignerOptions"
                  v-model="value.defaultX509Signer"
                  item-text="key"
                  class="font-weight-medium"
                  item-value="value"
                  @input="setDefaultSignerProtocol"
                  label="Select default X509 signer"
                  single-line
                ></v-select>
              </v-col>
              <v-col cols="12" md="12">
                <v-select
                  persistent-hint
                  hint="Slots"
                  required
                  :items="slots"
                  v-model="value.hw"
                  item-text="key"
                  class="font-weight-medium"
                  item-value="value"
                  @input="setHardwareModule"
                  label="Select hardware module slot"
                  single-line
                ></v-select>
              </v-col>
            </v-row>

            <v-row>
              <v-col md="12" cols="12">
                <v-alert text color="blue" v-if="loading">
                  <v-progress-circular
                    indeterminate
                    v-if="loading"
                    color="blue darken-1"
                  ></v-progress-circular>
                  {{ operationType }} ...
                </v-alert>
                <v-alert text :color="alertType" v-if="alertMessage">
                  {{ alertMessage }}
                </v-alert>
              </v-col>
            </v-row>
          </v-form>
        </v-card-text>

        <v-card-actions>
          <v-spacer></v-spacer>
          <v-btn color="blue darken-1" text @click="show = false">Cancel</v-btn>
          <v-btn color="blue darken-1" text @click="change">OK</v-btn>
        </v-card-actions>
      </v-card>
    </v-dialog>
  </v-content>
</template>
<script lang="ts">
import { TypedRFE, TasaISC, ISC } from 'xdvplatform-wallet/src';
import { Component, Prop, Vue, Watch } from 'vue-property-decorator';

import { SmartCardConnectorPKCS11 } from '../shared/SmartCardConnector';
import TransportWebUSB from '@ledgerhq/hw-transport-webusb';
import Eth from '@ledgerhq/hw-app-eth';
import {
  KeystoreIndex,
  Capability,
  DIDSigner,
  X509Signer,
} from '../shared/KeystoreIndex';
import { Session } from '../shared/Session';
import { async } from 'rxjs/internal/scheduler/async';
import {
  PKCS11ToolBindingsConfig,
  PKCS11ToolBindings,
} from '../shared/PKCS11ToolBindings';

@Component({
  name: 'xdv-link-external-keystore',
  props: ['show', 'value', 'keystore'],
})
export default class LinkExternalKeystore extends Vue {
  value: {
    type: 'walletconnect' | 'xdv' | 'pkcs11' | 'ledger';
    defaultDIDSigner: DIDSigner;
    defaultX509Signer: X509Signer;
    hw: '';
  };
  show;
  keystore: KeystoreIndex;

  defaultDIDSignerOptions = [
    {
      key: 'Ledger',
      value: DIDSigner.Ledger,
    },
    // {
    //   key: 'WalletConnect',
    //   value: DIDSigner.Walletconnect,
    // },
    {
      key: 'XDV',
      value: DIDSigner.XDV,
    },
  ];
  defaultX509SignerOptions = [
    {
      key: 'PKCS#11',
      value: X509Signer.PKCS11,
    },
    // {
    //   key: 'PKCS#12',
    //   value: X509Signer.PKCS12,
    // },
    {
      key: 'XDV Self Signed',
      value: X509Signer.XDV,
    },
  ];
  loading = false;
  alertMessage: string = '';
  alertType: string = '';
  pin = '';
  showPassword = '';
  isLoginSmartCardReady: boolean;
  validations = { password: undefined };
  showSCLogin = false;
  slots = [];

  async setHardwareModule() {}

  async loginSmartCard() {
    try {
      // // login
      // const config = this.pkcs11Config;
      // config.isSimulator = true;
      // await smartCardConnector.login(this.pin);
      this.isLoginSmartCardReady = true;
      // this.validations.password = 'Invalid PIN';
      // await this.linkSmartCard();
      this.showSCLogin = false;
    } catch (e) {
      console.log(e);
      this.validations.password = false;
      this.isLoginSmartCardReady = false;
    }
  }

  async linkLedger() {
    const transport = await TransportWebUSB.open();
    const eth = new Eth(transport);

    const addr = await eth.getAddress("44'/60'/0'/0/0");
  }

  async linkSmartCard() {
    const smartCardConnector = new SmartCardConnectorPKCS11();
    // login
    this.slots = await smartCardConnector.getSlots();
    await this.onSelectHWModule(this.slots[0], null);
  }

  @Watch('value.hw')
  async onSelectHWModule(current, old) {
    let keystore;
    if (this.value.defaultX509Signer === X509Signer.PKCS11) {
      keystore = {
        ...this.keystore,
        linkedExternalKeystores: {
          ...this.keystore.linkedExternalKeystores,
          pkcs11: {
            tokenIndex: this.slots.findIndex(i => i.key === current.slotDescription),
            capability: Capability.Sign,
          },
        },
      };
    }
    this.keystore = keystore;
    await Session.setWalletRefs(keystore, true);
  }

  async mounted() {}

  async setDefaultSignerProtocol() {
    const keystore = {
      ...this.keystore,
      defaultDIDSigner: this.value.defaultDIDSigner,
      defaultX509Signer: this.value.defaultX509Signer,
    };

    await Session.setWalletRefs(keystore, true);
  }
  async change() {
    await this.setDefaultSignerProtocol();
    this.$emit('input', this.value);
  }

  @Watch('value.defaultX509Signer')
  async onX509SignerChange(current: X509Signer) {
    if (current === X509Signer.PKCS11) {
      await this.linkSmartCard();
    } else {
      const keystore = {
        ...this.keystore,
        defaultDIDSigner: this.value.defaultDIDSigner,
        defaultX509Signer: this.value.defaultX509Signer,
      };

      await Session.setWalletRefs(keystore, true);
    }
  }

  @Watch('value.defaultDIDSigner')
  async onDIDSignerChange(current: DIDSigner) {
    if (current === DIDSigner.Ledger) {
      await this.linkLedger();
    } else {
            const keystore = {
        ...this.keystore,
        defaultDIDSigner: this.value.defaultDIDSigner,
        defaultX509Signer: this.value.defaultX509Signer,
      };

      await Session.setWalletRefs(keystore, true);
    }
  }
}
</script>
