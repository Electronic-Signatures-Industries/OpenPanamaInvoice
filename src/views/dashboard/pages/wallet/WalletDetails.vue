<template>
  <v-expansion-panel-content>
    <v-list>
      <v-list-item
        v-for="(item, i) in this.items"
        two-line
        :key="i"
      >
        <v-list-item-content>
          <v-list-item-title>{{ item.title }}</v-list-item-title>
          <v-list-item-subtitle>{{ item.subtitle }}</v-list-item-subtitle>
        </v-list-item-content>
      </v-list-item>
    </v-list>
  </v-expansion-panel-content>
</template>

<script lang="ts">
import { Component, Prop, Vue } from "vue-property-decorator";
import { KeystoreIndex } from "../shared/KeystoreIndex";

interface WalletDetailsItem {
  title: String;
  subtitle: String;
}

@Component({})
export default class WalletDetails extends Vue {
  @Prop()
  readonly didId: String;

  @Prop()
  readonly currentAddress: String;

  @Prop()
  readonly currentKeystore?: KeystoreIndex;

  get items(): WalletDetailsItem[] {
    return [{
      title: "DID",
      subtitle: this.didId
    }, {
      title: "Address",
      subtitle: this.currentAddress
    }, {
      title: "Linked to Smart Card",
      subtitle: this.getP11Name()
    }, {
      title: "Linked to P12",
      subtitle: this.getP12Name()
    }, {
      title: "Creation Date",
      subtitle: this.getKeystoreCreationDate()
    }]
  }

  getKeystoreCreationDate(): String {
    if (!this.currentKeystore) return '';
    return new Date(this.currentKeystore.created).toString();
  }

  getP11Name() {
    return this?.currentKeystore?.linkedExternalKeystores?.pkcs11?.tokenIndex || "no";
  }

  getP12Name() {
    if (this?.currentKeystore?.linkedExternalKeystores?.pkcs12) {
      return (
        this.currentKeystore.linkedExternalKeystores.pkcs12?.name ||
        "No name found"
      );
    }

    return "no";
  }
}
</script>