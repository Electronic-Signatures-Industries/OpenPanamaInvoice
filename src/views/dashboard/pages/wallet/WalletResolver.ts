import axios from 'axios';

const base = `http://localhost:3000`;

export class WalletResolver {
    constructor() {

    }

     static async  setPublicWalletRef(
         commonName: string,
         ref: string
     ) {
         return await axios.post(`${base}/domains/wallet`, {
             walletReference: ref,
             commonName
         });
    }
}