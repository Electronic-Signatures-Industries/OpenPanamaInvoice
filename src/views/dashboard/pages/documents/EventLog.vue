<template>
  <v-card>
    <v-card-title>Event log</v-card-title>

    <v-card-text>
      <v-list>
        <v-list-item>
          <v-list-item-content>
            <v-list-item-title>Nombre</v-list-item-title>
            <v-list-item-subtitle>{{ this.name }}</v-list-item-subtitle>
          </v-list-item-content>
        </v-list-item>

        <v-list-item>
          <v-list-item-content>
            <v-list-item-title>Archivo</v-list-item-title>
            <v-list-item-subtitle>{{ this.contentType }}</v-list-item-subtitle>
          </v-list-item-content>
        </v-list-item>
        
        <v-list-item>
          <v-list-item-content>
            <v-list-item-title>Timestamp</v-list-item-title>
            <v-list-item-subtitle>{{ this.timestamp }}</v-list-item-subtitle>
          </v-list-item-content>
        </v-list-item>
      </v-list>
    </v-card-text>

    <v-card-actions>
      <v-spacer />
      <v-btn
        v-if="this.documentMetadata"
        @click="onDownload"
        data-cy="download-button"
        text
        >
          <v-icon>mdi-download</v-icon>
          Descargar Archivo
      </v-btn>
    </v-card-actions>
  </v-card>
</template>

<script lang="ts">
import Vue from 'vue'
import { Component, Emit, Prop } from 'vue-property-decorator';
import Moment from "moment";
import { DocumentMetadata } from '@/views/dashboard/pages/wallet/IPFSManager';

@Component({})
export default class EventLog extends Vue {
  @Prop()
  readonly documentMetadata: DocumentMetadata | null;

  @Emit('download')
  onDownload() {
    return this.documentMetadata;
  }

  get timestamp(): string {
    return this.documentMetadata ? Moment(1000 * this.documentMetadata.timestamp).toString() : '';
  }

  get name(): string {
    return this.documentMetadata?.name ?? '';
  }

  get contentType(): string {
    return this.documentMetadata?.contentType ?? '';
  }
}
</script>