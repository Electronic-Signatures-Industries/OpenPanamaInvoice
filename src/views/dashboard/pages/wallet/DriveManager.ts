import { DID } from 'dids';
import { forkJoin } from 'rxjs';
import { DIDManager } from './DIDManager';
import { IPFSManager } from './IPFSManager';

export class DriveManager {
    constructor(
        private ipfs: IPFSManager,
        private did: DID        
    ) {

    }

    async createDocumentSet(files: File[]){
        // get promises of signed objects
        const pushFiles = files.map(async (f) => {
            return await this.ipfs.addSignedObject(this.did, f);
        });

        // run promises
        const items = await forkJoin(pushFiles).toPromise();
        
        // get indexed of ipfs
        const parentindex = await this.ipfs.addIndex(this.did, items);

        return parentindex;
    }

    async appendDocumentSet(files: File[]) {

        // get promises of signed objects
        const pushFiles = files.map(async (f) => {
            return await this.ipfs.addSignedObject(this.did, f);
        });

        // response with indexes (IPFS cid's)
        const res = await this.ipfs.setCurrentNode(
            parentindex,
        );

        return res.name;
    }
}