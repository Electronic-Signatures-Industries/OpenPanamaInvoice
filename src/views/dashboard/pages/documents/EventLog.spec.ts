import { createLocalVue, mount, Wrapper } from "@vue/test-utils";
import Vuetify from "vuetify";
import { DocumentMetadata } from "@/views/dashboard/pages/wallet/IPFSManager";
import EventLog from "./EventLog.vue";

const documentMetadata: DocumentMetadata = {
  name: "Mocked Document Name",
  contentType: "application/json",
  timestamp: 1000,
};

describe("EventLog.vue", () => {
  let component: Wrapper<EventLog>;

  beforeEach(() => {
    component = mount(EventLog, {
      localVue: createLocalVue(),
      vuetify: new Vuetify(),
    });
  });

  describe("snapshots", () => {
    it("should match when document is null", () => {
      expect(component).toMatchSnapshot();
    });

    it("should match when document is defined", async () => {
      component.setProps({ documentMetadata });
      await component.vm.$nextTick();

      expect(component).toMatchSnapshot();
    });
  });

  it("should emit the document metadata when the link is pressed", async () => {
    component.setProps({ documentMetadata });
    await component.vm.$nextTick();

    const button = component.find("[data-cy=download-button]");
    button.trigger("click");

    expect(component.emitted("download")[0][0]).toMatchSnapshot();
  });
});
