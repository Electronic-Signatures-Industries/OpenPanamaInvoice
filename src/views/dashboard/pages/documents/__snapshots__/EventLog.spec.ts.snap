// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventLog.vue should emit the document metadata when the link is pressed 1`] = `
Object {
  "contentType": "application/json",
  "name": "Mocked Document Name",
  "timestamp": 1000,
}
`;

exports[`EventLog.vue snapshots should match when document is defined 1`] = `
<div class="mx-auto v-card v-sheet theme--light">
  <div class="v-card__title">Event log</div>
  <div class="v-card__text">
    <div role="list" class="v-list v-sheet theme--light">
      <div tabindex="-1" role="listitem" class="v-list-item theme--light">
        <div class="v-list-item__content">
          <div class="v-list-item__title">Nombre</div>
          <div class="v-list-item__subtitle">Mocked Document Name</div>
        </div>
      </div>
      <div tabindex="-1" role="listitem" class="v-list-item theme--light">
        <div class="v-list-item__content">
          <div class="v-list-item__title">Archivo</div>
          <div class="v-list-item__subtitle"><a data-cy="download-button">
              application/json
            </a></div>
        </div>
      </div>
      <div tabindex="-1" role="listitem" class="v-list-item theme--light">
        <div class="v-list-item__content">
          <div class="v-list-item__title">Timestamp</div>
          <div class="v-list-item__subtitle">Thu Jan 01 1970 00:16:40 GMT+0000</div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EventLog.vue snapshots should match when document is null 1`] = `
<div class="mx-auto v-card v-sheet theme--light">
  <div class="v-card__title">Event log</div>
  <div class="v-card__text">
    <div role="list" class="v-list v-sheet theme--light">
      <div tabindex="-1" role="listitem" class="v-list-item theme--light">
        <div class="v-list-item__content">
          <div class="v-list-item__title">Nombre</div>
          <div class="v-list-item__subtitle"></div>
        </div>
      </div>
      <div tabindex="-1" role="listitem" class="v-list-item theme--light">
        <div class="v-list-item__content">
          <div class="v-list-item__title">Archivo</div>
          <div class="v-list-item__subtitle"><a data-cy="download-button">

            </a></div>
        </div>
      </div>
      <div tabindex="-1" role="listitem" class="v-list-item theme--light">
        <div class="v-list-item__content">
          <div class="v-list-item__title">Timestamp</div>
          <div class="v-list-item__subtitle"></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
